presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestReorderWindows.java

```diff
@@ -246,9 +246,8 @@ public void testNotReorderAcrossNonPartitionFilter()
                         window(windowMatcherBuilder -> windowMatcherBuilder
                                         .specification(windowA)
                                         .addFunction(functionCall("avg", commonFrame, ImmutableList.of(QUANTITY_ALIAS))),
-                                filter(
-                                        RECEIPTDATE_ALIAS + " IS NOT NULL",
-                                        project(
+                                project(
```

Comment:
where did projection came from?

Reply:
The projection was in this test before, it's been moved above the filter because I added the PredicatePushDown optimizer.

This projection is dropping symbols that were needed for the window, but that are not needed any more. In this specific case, it's dropping `discount`, `tax`, and `receiptdate`.